src/ucp/api/ucp.h

```diff
@@ -1172,6 +1224,73 @@ typedef struct ucp_worker_params {
 } ucp_worker_params_t;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP endpoint performance evaluation request attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the request for performance estimation of a particular endpoint.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_perf_param_field_t.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Message size to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    size_t            message_size;
+
+    /**
+     * Local memory type to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t local_memory_type;
```

Comment:
the field is not user by the implementation and not tested by unit test. Why we introduce it now ?

Reply:
We want to allow applications to pass more information to UCP, so future implementations of the UCX library will be able to make a more accurate performance estimation. IMO it is better than requiring application developers to keep updating their codes and pass additional parameters.

Reply:
But we are ignoring this parameter and don't test it..I don't see why would anybody use it.

Reply:
do you think it would be better to remove this from API until it's really implemented?

Reply:
If we don't implement it and test it, it makes more sense to remove it. It can be easy added later on.